lib/i18n/tasks/scanners/relative_keys.rb

```diff
@@ -42,7 +42,7 @@ def path_root(path, roots)
         def prefix(normalized_path, calling_method: nil)
           file_key       = normalized_path.gsub(%r{(\.[^/]+)*$}, '').tr(File::SEPARATOR, DOT)
           calling_method = calling_method.call if calling_method.respond_to?(:call)
-          if calling_method&.present?
+          if PatternScanner.new.key_relative_to_method?(normalized_path)
```

Comment:
1. It'd probably best to move `key_relative_to_method?` here instead of instantiating a `PatternScanner`.
2. If `calling_method` is `nil`, the result will have a trailing dot, which is probably wrong / unintuitive. I wonder what i18n actually generates in this case.